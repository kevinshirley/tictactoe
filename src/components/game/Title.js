import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import logo from '../../assets/logo.svg';
import TextFieldGroup from '../common/TextFieldGroup';

class Title extends Component {
  constructor() {
		super();
		
    this.state = {
      newGameOpenModal: true,
      players: '',
      playerOne: '',
      playerTwo: '',
    };
    
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({[e.target.name]: e.target.value});
  }

  componentDidMount(prevProps) {
    if (this.props.players !== this.state.players) {
      let players = this.props.players;
      let playerOne = '';
      let playerTwo = '';
      if (players) {
        playerOne = (players.playerOne ? players.playerOne : '');
        playerTwo = (players.playerTwo ? players.playerTwo : '');
      }
      this.setState({
        playerOne: playerOne,
        playerTwo: playerTwo,
      });
    }
  }

  render() {
		return (
			<section className="game-title">
        <div className="overlay">

          <div className="logo">
            <Link to={'/'}><img src={logo} alt="Tic Tac Toe" /></Link>
          </div>

          <div className="players">
            <TextFieldGroup 
              placeholder="Enter the player name"
              name="playerOne"
              type="text"
              value={this.state.playerOne}
              onChange={this.onChange}
              error={''}
              id="in-game-player-1"
              htmlFor="in-game-player-1"
              icon="people"
              label="Player 1"
              disabled={true}
            />

            <TextFieldGroup 
              placeholder="Enter the player name"
              name="playerTwo"
              type="text"
              value={this.state.playerTwo}
              onChange={this.onChange}
              error={''}
              id="in-game-player-2"
              htmlFor="in-game-player-2"
              icon="people"
              label="Player 2"
              disabled={true}
            />
          </div>

        </div>
			</section>
		);
	}
}

export default Title;