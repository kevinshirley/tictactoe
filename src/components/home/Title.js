import React, { Component } from 'react';
import logo from '../../assets/logo.svg';

class Title extends Component {
  render() {
		return (
			<section className="home-title">

        <img src={logo} alt="Tic Tac Toe" />
				
			</section>
		);
	}
}

export default Title;