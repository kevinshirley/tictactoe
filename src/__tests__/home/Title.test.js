import React from 'react';
import { shallow } from 'enzyme';
import Title from '../../components/home/Title';

describe('Title', () => {
  it('should render correctly in "debug" mode', () => {
    const title = shallow(<Title debug />);
  
    expect(title).toMatchSnapshot();
  });
});