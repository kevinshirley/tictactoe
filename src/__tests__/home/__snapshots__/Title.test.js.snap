// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render correctly in "debug" mode 1`] = `
<section
  className="home-title"
>
  <img
    alt="Tic Tac Toe"
    src=""
  />
</section>
`;
